webrender/src/frame.rs

```diff
@@ -319,6 +319,32 @@ impl Frame {
                 continue;
             }
 
+           let mut delta:Point2D<f32> = match scroll_location {
+            ScrollLocation::Delta(delta) => delta,
+            ScrollLocation::Start => {
+                if layer.scrolling.offset.y.round() == 0.0 {
+                    // Nothing to do.
+                    return false;
+                }
+
+                layer.scrolling.offset.y = 0.0;
+                return true;
+            },
+            ScrollLocation::End => {
+                let end_pos = -layer.content_size.height +
+                                                 (layer.local_viewport_rect.size.height);
```

Comment:
This is probably clearer as layer.local_viewport_rect.size.height - layer.content_size.height